### ISSUE TITLE
Fails to resume from hibernation

### ISSUE THREAD

**Issue by Sinon434** [veracrypt/VeraCrypt#739]
I can no longer recover Windows from a hibernated state after using system partition encrypting for some time.

From what read in some documentation, VeraCrypt should allow me to use hibernation the way one normally would on a Windows system.

After having my system partition encrypted for a few days and hibernation working fine, Windows suddenly began failing to recover from hibernation every time it hibernates. Event Viewer provides this error in system logs. "Windows failed to resume from hibernate with error status 0xC000009A."

I unfortunately don't have many clues why this is occurring. I do recall updating Windows a few days after encrypting, however not sure if that effects anything.

I’m currently running Windows 10 Home x64 Build 18364.
VeraCrypt 1.24-Update7.
Legion Y540 gaming laptop.

**Comment by Sinon434** [comment]
I have found a temporary fix to this issue. I found a discussion thread with a similar problem where a system that had Windows updated was hibernated, then was not able to boot into Windows, though fixed by re-encrypting the system partition.

I performed this and hibernation is working again as it should. Though this unfortunately means I’ll have to re-encrypt the system partition after every update for hibernation to continue to work, so I wouldn't consider this issue solved.

**Comment by idrassi** [comment]
Thanks for the report, and sorry for the very late follow-up.

The error code `0xC000009A` is `STATUS_INSUFFICIENT_RESOURCES`. In this context, Windows is failing while restoring the hibernation image because required resources are not available. This is different from a VeraCrypt pre-boot authentication or bootloader failure.

VeraCrypt system encryption is designed to support Windows hibernation when the hibernation file is located on the encrypted system partition. The main exceptions are configurations where VeraCrypt intentionally disables hibernation, such as RAM encryption, or hidden OS setups with an extra boot partition. Fast Startup should also be disabled.

The decrypt/re-encrypt workaround you found probably refreshed the bootloader / Windows upgrade state, but it shouldn't be needed after every Windows update. Since VeraCrypt `1.24-Update7`, several related changes have been added, including Fast Startup handling, Windows Feature Update bootloader/SetupConfig repair, and driver dump-filter fixes.

At this point this old report isn't enough to identify an actionable bug in current VeraCrypt. I will close it for now. If this still happens with a current VeraCrypt version, please open a new issue or comment with:
- exact VeraCrypt version
- exact Windows version/build
- UEFI vs legacy boot, Secure Boot state, BIOS/firmware version
- whether RAM encryption and Fast Startup are enabled
- RAM size and memory/commit usage just before hibernation
- whether it fails only after Windows Feature Updates or also during normal hibernation
- Event Viewer entries around the failed resume
